### PR TITLE
KOGITO-9145 Added information about kogito-addons-quarkus-microprofile-config-service-catalog

### PR DESCRIPTION
**Pull request by hbelmiro**
**JIRA:** https://issues.redhat.com/browse/KOGITO-9145

This PR depends on:

- https://github.com/kiegroup/kogito-runtimes/pull/3179

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [x] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 1]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
 
 If the URI pattern is not found in the application properties, then discovery is not triggered. However, the scanning is performed anyway. Therefore, a short time is required to go through the startup scan.
 
+=== Available service discovery implementations
+
+Currently, there are two implementations available.
+
 You can enable the Kubernetes service discovery by adding a service discovery implementation to the application's dependencies as shown in the following Maven dependencies:
 
+==== `org.kie.kogito:kogito-addons-quarkus-fabric8-kubernetes-service-catalog`
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:282]: Can we use the name of the Add-on instead?

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 2]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
...
+
+When enabled, it uses the Kubernetes Java API to discover the services.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:284]: Can you improve this description by saying that's ideally used without the operator in self-managed scenarios? (when users build and deploy workflows on their own).

Comment (ricardozanini) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:284]: Later, I'll improve this by adding a link to the operator's guide (that we don't have it yet)

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 3]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
...
+    <artifactId>kogito-addons-quarkus-kubernetes</artifactId>
+</dependency>
+<dependency>
     <groupId>org.kie.kogito</groupId>
     <artifactId>kogito-addons-quarkus-fabric8-kubernetes-service-catalog</artifactId>
 </dependency>
 ----
 
-=== Available service discovery implementations
+==== `org.kie.kogito:kogito-addons-quarkus-microprofile-config-service-catalog`
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:299]: Same here about the description and expansion on the usage scenario.

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc (hunk 1 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc]: Done.

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc (hunk 2 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc]: Done.

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc (hunk 3 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc]: Done.

**Review by ricardozanini — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 4]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
 
 If the URI pattern is not found in the application properties, then discovery is not triggered. However, the scanning is performed anyway. Therefore, a short time is required to go through the startup scan.
 
+=== Available service discovery implementations
+
+Currently, there are two implementations available.
+
 You can enable the Kubernetes service discovery by adding a service discovery implementation to the application's dependencies as shown in the following Maven dependencies:
 
+==== Fabric8 Kubernetes Service Catalog
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:282]:
```suggestion
==== Fabric8 Kubernetes service catalog
```

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 5]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
...
+    <artifactId>kogito-addons-quarkus-kubernetes</artifactId>
+</dependency>
+<dependency>
     <groupId>org.kie.kogito</groupId>
     <artifactId>kogito-addons-quarkus-fabric8-kubernetes-service-catalog</artifactId>
 </dependency>
 ----
 
-=== Available service discovery implementations
+==== MicroProfile Config Kubernetes Service Catalog
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:299]:
```suggestion
==== MicroProfile Config Kubernetes service catalog
```

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc [hunk 6]
```diff
@@ -273,25 +273,45 @@ When using the Kubernetes service discovery feature, you need to balance if your
...
+
+This implementation retrieves information from the application's configuration, making it valuable in scenarios where users rely on the operator to provide the necessary information from the hosting platform."
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:301]:
```suggestion
This implementation retrieves information from the application's configuration, making it valuable in scenarios where users rely on the operator to provide the necessary information from the hosting platform.
```

**Review by kaldesai — APPROVED** [review]
@hbelmiro Very minor suggestions. Otherwise, your content looks good to me. Thank you!

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc (hunk 4 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:282]: That's the name of the module. So, I think uppercase is fine.

File: serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc (hunk 5 above)
Comment (hbelmiro) [reply serverlessworkflow/modules/ROOT/pages/cloud/quarkus/kubernetes-service-discovery.adoc:299]: Same as above.

**Review by radtriste — APPROVED** [review]

**Pull request merged**